server/server.js

```diff
@@ -183,6 +183,113 @@ app.post('/api/tripUpdate', (req, res) => {
     });
 });
 
+app.post('/api/userUpdate', (req, res) => {
```

Comment:
This method is maybe too long. Would be great to refactor it to use some helper functions in order to do not repeat code and make it more readable and clean...